static/autopilot.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2022 k0s authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package static
+
+import (
+	"bytes"
+	"embed"
+	"fmt"
+	"path"
+
+	extensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+const (
+	embeddedManifestPath = "manifests/autopilot.k0sproject.io/v1beta2"
+)
+
+var (
+	//go:embed manifests/autopilot.k0sproject.io/v1beta2/*.yaml
+	crds embed.FS
+)
+
+// LoadCustomResourceDefinitions extracts all of the embedded CRDs into a map, keyed
+// by the real CRD name. Failing to load or parse a CRD will result in an error.
+func LoadCustomResourceDefinitions() (map[string]string, error) {
```

Comment:
This code made sense with autopilot running standalone -- the CRD was embedded in the binary and applied on startup. Embedding this into `k0s` now, this should probably be done at `pkg/component/controller/crd.go`, or maybe even better in the new component that we add for autopilot-controllers (a different PR).

Either way, you can easily ignore this + manifests from this change, and remove the `LoadCustomResourceDefinitions()` usage.

Reply:
Removed

Reply:
The autopilot manifests are still in this diff, they can be removed.

Reply:
They are needed exactly for `pkg/component/controller/crd.go` and are used in https://github.com/k0sproject/k0s/pull/1739